index.js

```diff
@@ -143,6 +143,11 @@ function filter(patterns, opts) {
   }
 
   patterns = utils.arrayify(patterns);
+  var len = patterns.length, i = 0;
+  while (i < len) {
+    patterns[i] = matcher(patterns[i++], opts);
```

Comment:
Which confusion? Order of patterns or length of array aren't changed. I just reassigned variables